instant-distance/src/contiguous.rs

```diff
@@ -0,0 +1,240 @@
+use std::marker::PhantomData;
+
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    types::{LayerId, Meta, INVALID},
+    Builder, Point, PointId,
+};
+
+pub trait PointDataSource: Sync {
+    fn decompose(&self) -> Vec<f32>;
+    fn stride() -> usize;
```

Comment:
I previously suggested you make `STRIDE` an associated const, any particular reason you didn't follow up with that?

Reply:
In actuality I did have an associated const at one point. I migrated to a static function  during an exploratory change, but I'm happy to move back to a const.